webapp/components/interop.js

```diff
@@ -172,10 +174,9 @@ class InteropDataManager {
 
         let summaryScore = testScore / numFocusAreas;
 
-        // TODO: get the investigation score at this date.
-        if (this.year === '2022') {
-          summaryScore = summaryScore * 0.9;
-        }
+        // Handle investigation scoring if applicable.
+        summaryScore = this.#addInvestigationScore(summaryScore, date);
```

Comment:
The naming of `addInvestigationScore` and `summaryScore` bugs me a little bit, because this doesn't merely add to `summaryScore`, it also scales it by 0.9 (for 2022).

The problem is that we need the `investigationWeight` to know how to scale `summaryScore`. But I think it might be clearer like this:

```suggestion
        const [investigationScore, investigationWeight] = this.#getInvestigationScoreAndWeight(date);
```

And then compute a `summaryScore` here after the call. I think that'd make it easier to follow how the total score is computed from the individual parts.